### PR TITLE
Update score function

### PR DESCRIPTION
**Pull request by oooo26**
Update some score function in Python package:
- logistic & multinomial: predition accuracy
- linear & multi-task: [R^2 score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.r2_score.html#sklearn.metrics.r2_score)
- poisson: [D^2 score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.d2_tweedie_score.html#sklearn.metrics.d2_tweedie_score)

### CODE REVIEW

**Comment by oooo26** [comment]
Fix the strange "test_score" in #415.
